django/thunderstore/core/utils.py

```diff
@@ -70,11 +70,22 @@ def capture_exception(*args, **kwargs):
     capture_sentry_exception(*args, **kwargs)
 
 
-def make_full_url(request: Optional[HttpRequest], path: str):
+def make_full_url(
+    request: Optional[HttpRequest],
+    path: str,
+    site=None,
```

Comment:
The modifications to this method honestly seem like they should be two separate functions, with a 3rd base function, where:

Function 1 would have the base URL building implementation
Function 2 would accept a request object and use Function 1 to build an URL based on it
Function 3 would accept a site object and use Function 1 to build an URL based on it